solidity/random-beacon/tasks/mint.ts

```diff
@@ -0,0 +1,39 @@
+import { task, types } from "hardhat/config"
+
+import type { BigNumber } from "ethers"
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+
+task("mint", "Mints and approves T tokens for staking")
+  .addParam("owner", "Stake Owner address", undefined, types.string)
+  .addOptionalParam("amount", "Stake amount", 1_000_000, types.int)
+  .setAction(async (args, hre) => {
+    await setup(hre, args)
+  })
+
+async function setup(
+  hre: HardhatRuntimeEnvironment,
+  args: {
+    owner: string
+    amount: BigNumber
+  }
+) {
+  const { ethers, helpers } = hre
+  const { owner, amount } = args
+
+  const { deployer } = await helpers.signers.getNamedSigners()
+  const { to1e18 } = helpers.number
+  const t = await helpers.contracts.getContract("T")
+  const staking = await helpers.contracts.getContract("TokenStaking")
+
+  const toMintAndApprove = to1e18(amount)
+
+  console.log(
+    `Minting ${toMintAndApprove.toString()} T for the ${owner} and approving for staking ...`
+  )
+
+  await (await t.connect(deployer).mint(owner, toMintAndApprove)).wait()
```

Comment:
Why do we need to use the double await? Is the first call blocking on `await` until the transaction is mined or until it's submitted to the mempool?

Reply:
Geth is way slower in terms of mining compared to Hardhat. We must wait until the `mint` transaction resolves and is included in the chain. If we do not `wait()`, then `approve()` will fail, cause it won't see any minted tokens.